creusot/src/ctx.rs

```diff
@@ -309,7 +309,10 @@ impl<'tcx, 'sess> TranslationCtx<'tcx> {
     }
 
     fn mk_opacity(&self, item: DefId) -> Opacity {
-        if !matches!(util::item_type(self.tcx, item), ItemType::Predicate | ItemType::Logic) {
+        if !matches!(
+            util::item_type(self.tcx, item),
+            ItemType::Predicate | ItemType::Logic | ItemType::Ghost
```

Comment:
We should probably add an `is_ghost` or similar function on `item_type`, these kinds of checks are frequent.

Reply:
YEah... (Btw, this was an unsoundness where we could implement a logic method with a ghost method.